### PR TITLE
feat(set-pat): Implementing setting PAT up functionality

### LINKED ISSUE #107: Implement frontend for setting a GitHub PAT
image

### PR DESCRIPTION
**Pull request by german-mergestat**
Implementing setting PAT up functionality. (Resolves #107)

valid-token

token-saved

- [x] Token validation
- [x] Swapping the position of `GitHub Authentication`
- [x] Saving token in data base
- [x] Unit Testing (token valid, token invalid, token saved in db)

### CODE REVIEW

**Comment by patrickdevivo** [comment]
FYI seeing this warning in the console (https://www.chromium.org/developers/design-documents/create-amazing-password-forms/):

image

File: ui/src/views/repositories/hooks/useSetPat.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const [showValidation, setShowValidation] = useState(false)
+  const [isTokenValid, setIsTokenValid] = useState(false)
+  const [pat, setPAT] = useState('')
+
+  const [savePAT] = useMutation(SET_PAT, {
+    onError: (error: ApolloError) => {
+      showErrorAlert(error.message)
+    },
+    onCompleted: () => {
+      showSuccessAlert('PAT saved')
```
Comment (patrickdevivo) [inline ui/src/views/repositories/hooks/useSetPat.ts:17]:
```suggestion
      showSuccessAlert('GitHub access token saved')
```

**Comment by patrickdevivo** [comment]
@german-mergestat I think the validate button should also make an HTTP call to the GitHub GraphQL API to validate that it "works" as well:

```
query { 
  viewer { 
    login
  }
}
```
`https://api.github.com/graphql`

See here: https://docs.github.com/en/graphql/guides/forming-calls-with-graphql#authenticating-with-graphql

File: ui/src/views/repositories/hooks/useSetPat.ts (hunk 1 above)
Comment (german-mergestat) [reply ui/src/views/repositories/hooks/useSetPat.ts]: Done [9f30294](https://github.com/mergestat/fuse/pull/177/commits/9f3029429697b88ed475cbd5f52e9bda6b2d439d)

**Review by patrickdevivo — APPROVED** [review]

**Pull request merged**